cola/widgets/diff.py

```diff
@@ -195,6 +195,7 @@ def __init__(self, parent):
 
     def set_diff(self, diff):
         parser = self.parser
+        parser.valid = True
```

Comment:
When reading this diff my first reaction was that this seemed wrong.  But, it actually is the correct state change, but this `parser.valid = True` should actually be done at the very beginning of `def parse()` as `self.valid = True`.  That way the caller here does not need to deal with it.